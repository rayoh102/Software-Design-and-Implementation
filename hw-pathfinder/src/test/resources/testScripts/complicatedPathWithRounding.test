# Find a path with Dijkstra's algorithm where we
# need to use rounding.

CreateGraph simpleGraph

AddNode simpleGraph A
AddNode simpleGraph B
AddNode simpleGraph C
AddNode simpleGraph D

AddEdge simpleGraph A B 10.0001
AddEdge simpleGraph A C 12.214141
AddEdge simpleGraph C D 1.31332
AddEdge simpleGraph B D 3.513432

FindPath simpleGraph A D
