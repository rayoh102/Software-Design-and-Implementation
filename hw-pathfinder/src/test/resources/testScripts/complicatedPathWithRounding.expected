# Find a path with Dijkstra's algorithm where we
# need to use rounding.

created graph simpleGraph

added node A to simpleGraph
added node B to simpleGraph
added node C to simpleGraph
added node D to simpleGraph

added edge 10.000 from A to B in simpleGraph
added edge 12.214 from A to C in simpleGraph
added edge 1.313 from C to D in simpleGraph
added edge 3.513 from B to D in simpleGraph

path from A to D:
A to B with weight 10.000
B to D with weight 3.513
total cost: 13.514